SystemServices/SystemServices.cpp

```diff
@@ -2010,6 +2011,38 @@ namespace WPEFramework {
             returnResponse(resp);
         }
 
+        /***
+         * @brief : Thunder API to Set System Time
+         * @param1[in]	: {"params":{"timestamp": "<string>"}}
+         * @param2[out]	: {"jsonrpc":"2.0","id":3,"result":{"success":<bool>}}
+         * @return		: Core::<StatusCode>
+         */
+        uint32_t SystemServices::setTimeFromUser(const JsonObject& parameters, JsonObject& response)
+        {
+            LOGINFOMETHOD();
+            bool result = true;
+            if (!parameters.HasLabel("timestamp"))
+            {
+                result = false;
+                response["message"] = "Missing timestamp parameter, please specify it (use ISO 8601)";
+            }
+
+            if (result)
+            {
+                const string timestamp = parameters["timestamp"].String();
+                Core::Time newTime(0);
+                newTime.FromISO8601(timestamp);
+                if (!newTime.IsValid()) {
+                    result = false;
+                    response["message"] = "timestamp not recoginized. Please use ISO 8601 format";
+                } else {
+                    Core::SystemInfo::Instance().SetTime(newTime);
```

Comment:
We shouldn't modify the system time directly. We should send a IARM message to System Time Manager. You can refer to https://gerrit.teamccp.com/#/c/461190/16/recipes-extended/rdkservices/files/0001-Added-Support-for-querying-and-publishing-the-Secure.patch for how to talk to System Time manager.